.github/workflows/ci.yml

```diff
@@ -0,0 +1,403 @@
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+      - name: Download nginx Images
+        uses: actions/download-artifact@v2
+        with:
+          name: docker-image-prometheus-exporter-file
+          path: ./tmp
+      - run: docker load --input ./tmp/image-prometheus-exporter-file.tar
+      - run: sudo chown -R 1000:1000 ./test/functional/web/tmp/ # Ensure we have the same uid:gid as our `app` docker user
+      - run: make test-prometheus-exporter-file-e2e
+  push-prometheus-exporter-file:
+    name: Push prometheus-exporter-file
+    if: (github.event_name == 'push' || github.event_name == 'schedule') && github.ref == 'refs/heads/master'
+    needs:
+      - test-http
+      - test-php
+      - test-prometheus-exporter-file
+      - scan-vulnerability-http
+      - scan-vulnerability-php
+      - scan-vulnerability-prometheus-exporter-file
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+      - name: Download Images
+        uses: actions/download-artifact@v2
+        with:
+          name: docker-image-prometheus-exporter-file
+          path: ./tmp
+      - run: docker load --input ./tmp/image*.tar
+      - run: make ci-docker-login push-prometheus-exporter-file
+        env:
+          DOCKER_USER: ${{ secrets.HUB_USERNAME }}
+          DOCKER_PASSWORD: ${{ secrets.HUB_PASSCODE }}
+  push-http:
+    name: Push nginx ${{ matrix.nginx }}
+    if: (github.event_name == 'push' || github.event_name == 'schedule') && github.ref == 'refs/heads/master'
+    needs:
+      - test-http
+      - test-php
+      - test-prometheus-exporter-file
+      - scan-vulnerability-http
+      - scan-vulnerability-php
+      - scan-vulnerability-prometheus-exporter-file
+      - generate-nginx-strategy
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        nginx: ${{ fromJson(needs.generate-nginx-strategy.outputs.nginx) }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Download Images
+        uses: actions/download-artifact@v2
+        with:
+          name: docker-image-http-${{ matrix.nginx }}
+          path: ./tmp
+      - run: docker load --input ./tmp/image*.tar
+      - run: make ci-docker-login push-http
+        env:
+          DOCKER_USER: ${{ secrets.HUB_USERNAME }}
+          DOCKER_PASSWORD: ${{ secrets.HUB_PASSCODE }}
+  push-php:
+    name: Push PHP ${{ matrix.php }} for ${{ matrix.type }} on Alpine ${{ matrix.alpine }}
+    if: (github.event_name == 'push' || github.event_name == 'schedule') && github.ref == 'refs/heads/master'
+    needs:
+      - test-http
+      - test-php
+      - test-prometheus-exporter-file
+      - scan-vulnerability-http
+      - scan-vulnerability-php
+      - scan-vulnerability-prometheus-exporter-file
+      - generate-alpine-strategy
+      - generate-php-strategy
+      - generate-php-type-strategy
```

Comment:
these are redundant because `test-php` already `needs` them.
there are probably a few unnecessary dependencies like this.